### PR TITLE
Fix OTG not loading file lists on Android >= N + refactoring

### LINKED ISSUE #1951: OTG storage not visible in Amaze
**Describe the bug**
Cannot see the OTG storage device in the drawer when plugged in.

**To Reproduce**
1. Plug in the USB storage with OTG cable
2. Open Amaze
3. Open drawer
4. OTG device not visible

**Expected behavior**
OTG device should be visible and accessible from within Amaze.

**Smartphone (please complete the following information):**
 - Device: Fairphone 3
 - OS: LineageOS 16.0 (9.0)
 - Rooted: Yes
 - Version: `release/3.5` branch

**Additional context**
Should be happening on Android >= N too.

### PR DESCRIPTION
**Pull request by TranceLove**
#### Issue tracker   
Addresses #1951 

#### Release  
Addresses release/3.5
  
#### Test cases
- [ ] Covered
  
#### Manual testing
- [x] Done  
  
If yes,  
- Device: Fairphone 3
- OS: LineageOS 16.0 (9.0)

#### Build tasks success  
Successfully running following tasks on local 
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

#### Additional Info

Changes:
- Allow USB device detection on Android >= N
- Add back missing dialog.show() in Drawer when asking user to open USB device root at Android DocumentsUI
- Moved SSH isDirectory() symlink checks to SshClientUtil
- combined MediaStore image/video query methods together + fix arguments

Known issue:
Amaze cannot detect USB device inserted while running. Debug mode saw Amaze did able to detect USB and put item drawer, but not at normal run.

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java [hunk 1]
```diff
@@ -786,29 +723,14 @@ public Void execute(SFTPClient client) {
...
+                      } catch (IOException ifBrokenSymlink) {
+                        Log.w(TAG, "IOException checking isDirectory(): " + info.getPath());
+                        continue;
                       }
-                      HybridFileParcelable f =
-                          new HybridFileParcelable(String.format("%s/%s", path, info.getName()));
-                      f.setName(info.getName());
-                      f.setMode(OpenMode.SFTP);
-                      f.setDirectory(isDirectory);
-                      f.setDate(info.getAttributes().getMtime() * 1000);
```
Comment (VishalNehra) [inline app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:807]: Not setting date/size/mode? in hybridFile?

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java [hunk 2]
```diff
@@ -825,12 +747,7 @@ public Void execute(SFTPClient client) {
         try {
           SmbFile smbFile = new SmbFile(path);
           for (SmbFile smbFile1 : smbFile.listFiles()) {
-            HybridFileParcelable baseFile = new HybridFileParcelable(smbFile1.getPath());
-            baseFile.setName(smbFile1.getName());
-            baseFile.setMode(OpenMode.SMB);
-            baseFile.setDirectory(smbFile1.isDirectory());
```
Comment (VishalNehra) [inline app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:831]: Not setting date/size/mode in hybridFile?

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java [hunk 3]
```diff
@@ -875,17 +792,15 @@ public Void execute(SFTPClient client) {
                       try {
                         for (RemoteResourceInfo info :
                             client.ls(SshClientUtils.extractRemotePathFrom(path))) {
+                          boolean isDirectory = false;
```
Comment (VishalNehra) [inline app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:795]: I see this method (listFiles) is expired. Maybe we can remove it's usage and point to forEachChildrenFile only?

**Review by VishalNehra — CHANGES_REQUESTED** [review]

**Comment by VishalNehra** [comment]
Might wanna remove PR_WIP label? ;)

**Comment by TranceLove** [comment]
> Might wanna remove PR_WIP label? ;)

But be reminded, plug and play may still not work after applying this PR, i.e. need to restart to see the OTG device in the drawer.

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java (hunk 2 above)
Comment (TranceLove) [reply app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:831]: Should be fine using HybridFileParcelable constructor for SMB files.

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java (hunk 1 above)
Comment (TranceLove) [reply app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:807]: Should be fine using HybridFileParcelable constructor for SFTP files.

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java (hunk 3 above)
Comment (TranceLove) [reply app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:795]: PrepareCopyTask,CopyNode still depends on it. Not dare to remove this at this moment.

**Comment by VishalNehra** [comment]
Can you make new HybridFileParcelable(smb) change in listFiles -> SFTP and forEachChildrenFile both places as well.

**Comment by TranceLove** [comment]
See if a5473eb fits.

File: app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java (hunk 1 above)
Comment (VishalNehra) [reply app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java:807]: Ahh, I overlooked, sorry my bad.

**Comment by VishalNehra** [comment]
Sorry my bad, overlooked a bit of the code.

**Review by VishalNehra — APPROVED** [review]

**Pull request merged**